oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,328 @@
+to avoid a failure during the initial LMS job. There were further MinIO name resolutions issues that precluded a fully
+working environment: there was no time to ascertain whether the cause is ultimately with Tutor or with the tested
+microk8s setup.
+
+It is mentioned at the top of the `Tutor Kubernetes documentation <https://docs.tutor.overhang.io/k8s.html>`__ that "the
+local Open edX install was designed such that users with no prior experience with system administration could still
+launch an Open edX platform. It is not the case for the installation method outlined here." While the Tutor project has
+clearly expended effort in getting it to work with Kubernetes, if it is to fully replace ``edx/configuration`` as the
+go-to community installation for production use, it is suggested that the Kubernetes documentation be extended with more
+detailed information on how to run and maintain Open edX in such an environment, and that Kubernetes support be made a
+first-class citizen.
+
+Note that it is perfectly possible to bypass Tutor entirely when it comes to cluster management. One could use its image
+creation engine to build the requisite Docker images, then manage the Kubernetes cluster with something like Helm: there
+are reports of `this exact thing being done
+<https://discuss.openedx.org/t/lets-talk-about-the-native-installation/3269/2>`__).
+
+
+Criterion 2. Automated deployment and testing
+.............................................
+
+Launching a preconfigured instance from scratch is simple, as Tutor can either `load configuration
+<https://docs.tutor.overhang.io/configuration.html#configuration>`__ from environment variables, from locally stored
+YAML files, or via command line parameters. Automating a launch is therefore just as simple as launching an
+``edx/configuration`` Ansible run.
+
+Nevertheless, it is not currently possible to run edx-platform tests in a Tutor-created container, so automation for
+this particular purpose will require work.
+
+
+Criterion 3. Support for the latest open release
+................................................
+
+Tutor supports the latest Open edX release, Koa, out of the box.  Furthermore, for at least the past two releases,
+development has been done concomitantly with the actual Open edX release itself.
+
+There's also built-in support for upgrading single servers, either `automatically
+<https://docs.tutor.overhang.io/local.html?highlight=upgrade#upgrading-from-earlier-versions>`__ or `manually
+<https://github.com/overhangio/tutor/blob/master/tutor/commands/local.py#L91>`__.  This, however, `does not apply to
+Kubernetes support <https://github.com/overhangio/tutor/blob/master/tutor/commands/k8s.py#L281-L288>`__.  While the
+corresponding single-server upgrade procedure can be used as an authoritative reference for how to achieve it manually
+on multi-server deployments, ideally there would be specific documentation on it.
```

Comment:
The manual instructions printed in the console should actually work for most people.